cardinal/evm/server.go

```diff
@@ -285,11 +286,12 @@ func (s *msgServerImpl) getSignerComponentForAuthorizedAddr(
 	wCtx := ecs.NewReadOnlyWorldContext(s.world)
 	q, err := wCtx.NewSearch(ecs.Exact(ecs.SignerComponent{}))
 	if err != nil {
-		return nil, err
+		return nil, eris.Wrap(err, "error creating search")
 	}
-	err = q.Each(wCtx, func(id entity.ID) bool {
+	err = errors.Join(err, q.Each(wCtx, func(id entity.ID) bool {
```

Comment:
why do we need to join the previous error and the error returned from `Each`? the initial error is already handled on line 289.

Reply:
You have to look ahead. The error by wCtx is overwritten by the error from the closure inside of q.Each. It's happening out of order here.

let me illustrate what was happening before:


```go
q, err := wCtx.NewSearch(ecs.Exact(ecs.SignerComponent{})) //1. Err is created here. 
	if err != nil {
		return nil, eris.Wrap(err, "error creating search") 
	}
	err = q.Each(wCtx, func(id entity.ID) bool { //3. error is overwritten here for a 3rd time. 
		var signerComp *ecs.SignerComponent
		signerComp, err = component.GetComponent[ecs.SignerComponent](wCtx, id) //2. error is overwritten here. 
		if err != nil {
			return false
```

Reply:
My code preserves the error created by GetComponent which was previously getting swallowed.

Reply:
ah i see. i think the code becomes a little unintuitive this way. i think separating the error from `.Each` and `.GetComponent` and handling them separately is a bit more straightforward. its a bit more code, but worthwhile imo for readability

Reply:
Agreed. This does break go convention though, but idc care for convention. I'll change it tmrw.

Reply:
i think its actually more go-convention-ish, since this way we will handle each error **explicitly**, which the gopher god prefers. with the join, we handle both each and an inner error as one error

Reply:
It's possible for both to occur though. Usually one error happens before another error so the second error doesn't get a chance to occur. Here we have a possibility of two errors that occured. When there are two errors, one will inevitably be swallowed if we don't join them. 

I think the problem is in the Each function. It doesn't allow errors to be propagated out of the closure. It should return a bool, err.

It's fine fow now I'll just do it on the agreed way.